python/pyspark/sql/readwriter.py

```diff
@@ -303,6 +330,40 @@ def load(
         if schema is not None:
             self.schema(schema)
         self.options(**options)
+
+        # Load a Python data source
+        if isinstance(self._format, Callable):  # type: ignore[arg-type]
+            # TODO(SPARK-45560): support load() with non-empty path.
+
+            # Create an instance of the data source.
+            data_source_cls = cast(Type[DataSource], self._format)
+            data_source = data_source_cls(self._options)
+
+            # Get schema of the data source
+            schema = self._schema or data_source.schema()
+            if isinstance(schema, str):
+                dt = _parse_datatype_string(schema)
```

Comment:
we will want to check for parsing failures here and return a reasonable error message.

Reply:
Yup. Added a TODO to make schema robust: [SPARK-45559](https://issues.apache.org/jira/browse/SPARK-45559)